lib/std/fs/path.zig

```diff
@@ -749,8 +749,18 @@ fn testResolvePosix(paths: []const []const u8, expected: []const u8) !void {
     return testing.expect(mem.eql(u8, actual, expected));
 }
 
+/// Returns the parant path of a directory path, or null when the root is reached.
+/// Asserts that the path has a directory component.
+pub fn parent(path: []const u8) ?[]const u8 {
+    const new_path = dirname(path) orelse unreachable;
```

Comment:
Strictly speaking this should be `orelse return "."`. When `dirname` returns null you either have a file name or an empty string so the parent is indeed `.`.